extensions/sql-database-projects/src/dialogs/publishOptionsDialog.ts

```diff
@@ -0,0 +1,161 @@
+
+			// Get the description of the selected option
+			this.disposableListeners.push(this.optionsTable.onRowSelected(async () => {
+				let row = this.optionsTable.selectedRows![0];
+				let label = this.optionsModel.optionsLabels[row];
+				await this.descriptionText.updateProperties({
+					value: this.optionsModel.getDescription(label)
+				});
+			}));
+
+			// Update deploy options value on checkbox onchange
+			this.disposableListeners.push(this.optionsTable.onCellAction!((rowState) => {
+				let checkboxState = <azdata.ICheckboxCellActionEventArgs>rowState;
+				if (checkboxState && checkboxState.row !== undefined) {
+					let label = this.optionsModel.optionsLabels[checkboxState.row];
+					this.optionsModel.optionsLookup[label] = checkboxState.checked;
+				}
+			}));
+
+			this.optionsFlexBuilder = view.modelBuilder.flexContainer()
+				.withLayout({
+					flexFlow: 'column'
+				}).component();
+
+			this.optionsFlexBuilder.addItem(this.optionsTable, { CSSStyles: { 'overflow': 'scroll', 'height': '65vh' } });
+			this.optionsFlexBuilder.addItem(this.descriptionHeading, { CSSStyles: { 'font-weight': 'bold', 'height': '30px' } });
+			this.optionsFlexBuilder.addItem(this.descriptionText, { CSSStyles: { 'padding': '4px', 'margin-right': '10px', 'overflow': 'scroll', 'height': '10vh' } });
+			await view.initializeModel(this.optionsFlexBuilder);
+			await this.optionsTable.focus();
+		});
+	}
+
+	// Update the default options to the options table area
+	private async updateOptionsTable(): Promise<void> {
+		let data = this.optionsModel.getOptionsData();
+		await this.optionsTable.updateProperties({
+			data: data,
+			columns: [
+				<azdata.CheckboxColumn>
+				{
+					value: 'Include',
+					type: azdata.ColumnType.checkBox,
+					action: azdata.ActionOnCellCheckboxCheck.customAction,
+					headerCssClass: 'display-none',
+					cssClass: 'no-borders align-with-header',
+					width: 50
+				},
+				{
+					value: 'Option Name',
+					headerCssClass: 'display-none',
+					cssClass: 'no-borders align-with-header',
+					width: 50
+				}
+			],
+			ariaRowCount: data.length
+		});
+	}
+
+	// Ok button click, will update the deployment options with selections
+	protected execute(): void {
+		this.optionsModel.setDeploymentOptions();
+		this.publish.setDeploymentOptions(this.optionsModel.deploymentOptions);
+		this.disposeListeners();
+	}
+
+	// Cancels the deploy options table
+	protected cancel(): void {
+		this.disposeListeners();
+	}
+
+	// Reset button click, resets all the options selection
+	private async reset(): Promise<void> {
+		let result = await this.publish.getDefaultDeploymentOptions();
+		this.optionsModel.deploymentOptions = result;
+
+		await this.updateOptionsTable();
+		this.optionsFlexBuilder.removeItem(this.optionsTable);
+		this.optionsFlexBuilder.insertItem(this.optionsTable, 0, { CSSStyles: { 'overflow': 'scroll', 'height': '65vh' } });
+	}
+
+	private disposeListeners(): void {
+		if (this.disposableListeners) {
```

Comment:
this.disposableListeners?.forEach...

But also you initialize this to an empty array so i don't see how it could be undefined regardless